### PR TITLE
EDQ 12.2.1.3.0 Docker Scripts

### PR DESCRIPTION
**Pull request by subhashsutrave**
Initial EDQ 12.2.1.3.0 Docker Scripts

### CODE REVIEW

File: OracleEDQ/12.2.1.3.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+# ORACLE DOCKERFILES PROJECT
+# --------------------------
+# This is the Dockerfile for Oracle Enterprise Data Quality 12.2.1.3.0
+#
+# Base image of this dockerfile is Oracle Java 8 docker image.
+#
+# Pre-requisite to build oracle Oracle Enterprise Data Quality 12.2.1.3.0 image is to have
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/Dockerfile]: Actually, the pre-requisite is to build the FMW Infrastructure image, not just the Java 8 image.

File: OracleEDQ/12.2.1.3.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#
+#
+# Pull base image
+# ---------------
+FROM oracle/fmw-infrastructure:12.2.1.3
+
+#
+# Maintainer
+# ----------
+#MAINTAINER
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/Dockerfile]: Why are you not providing a `MAINTAINER`? This needs to be provided so we know who to assign any potential issues or pull requests to in future.

File: OracleEDQ/12.2.1.3.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    DOMAIN_NAME="edq_domain" \
+    DOMAIN_ROOT="${DOMAIN_ROOT:-/u01/oracle/user_projects/domains}" \
+    ADMIN_PORT=7001 \
+    EDQ_PORT=8001
+
+#
+# Proxy needs to be set before running yum 
+# ----------------------------------------
+USER root
+#RUN useradd -b /u01 -d /u01/oracle -m -s /bin/bash oracle && \
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/Dockerfile]: Don't leave commented directives in your `Dockerfile`. If you don't need this line, remove it.

File: OracleEDQ/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+e.g.:
+
+        $ docker run --name EDQAS -it -p 7001:7001 --env-file ./12.2.1.3.0/edq.db_env.list -v /scratch/xyz:/u01/oracle/user_projects oracle/edq:12.2.1.3.0
+
+On success, the following is logged: `The configuration of Oracle Fusion Middleware completed successfully.`. The script then tails the domain log to stdout.
+
+
+### Starting another container to Start EDQ Managed Server
+
+Start a container to launch the Managed Server from the image created in prior step  The environment variables used to run the Managed Server image are defined in the file EDQServer.env.list.
```
Comment (Djelibeybi) [inline OracleEDQ/README.md]: Why are you requiring a file when you just have two variables? Why not just define them in the `Dockerfile`?

File: OracleEDQ/12.2.1.3.0/EDQServer.env.list [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ADMIN_HOST=<EDQ Admin Server Host>
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/EDQServer.env.list]: Can you not provide a default value, e.g. `localhost`?

File: OracleEDQ/12.2.1.3.0/EDQServer.env.list [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
+ADMIN_PORT=<EDQ Admin Server Port>
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/EDQServer.env.list]: Can you not provide the default port?

File: OracleEDQ/12.2.1.3.0/container-scripts/CreateEDQDomain.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python
+# Copyright (c) 2014-2017 Oracle and/or its affiliates. All rights reserved.
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+# Script to make simple EDQ 12.2.1 domain
+# ---------------------------------------
+execfile("/u01/oracle/container-scripts/edq.py")
+i = 1
+while i < len(sys.argv):
```
Comment (Djelibeybi) [inline OracleEDQ/12.2.1.3.0/container-scripts/CreateEDQDomain.py]: You really should be using `argparse` for this and not a huge `while` block: https://docs.python.org/2.7/library/argparse.html

The `argparse` module is available by default in the `oraclelinux:7-slim` container.

File: OracleEDQ/README.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+If using a 12.1.0.2 CDB/PDB database, ensure PDB is used when creating the schemas. CDB is not supported.
+
+An Oracle Database 12.1.0.2 container can be created from [OracleDatabase](https://github.com/oracle/docker-images/tree/master/OracleDatabase/dockerfiles/12.1.0.2). 
+Follow the instructions to create a 12.1.0.2-based Enterprise Edition database.
+
+Default installs of Oracle Database 12.2 are not supported - please see [Known Issues](#known-issues).
+
+## Oracle Enterprise Data Quality Docker Image Creation
+
+To build a EDQ image you start by building Oracle JDK and Oracle FMW Infrastructure images.
```
Comment (Djelibeybi) [inline OracleEDQ/README.md]: It's not a JDK image, it's a JRE image. I would replace "Oracle JDK" with "Oracle Java 8 (Server JRE)"

File: OracleEDQ/README.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+### Building the Oracle JDK (Server JRE) Image
```
Comment (Djelibeybi) [inline OracleEDQ/README.md]: This should be "Oracle Java 8 (Server JRE) Image"

File: OracleEDQ/README.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+### Building the Oracle Enterprise Data Quality Image
+
+Download the two binaries for [Oracle Enterprise Data Quality 12.2.1.3.0]( http://www.oracle.com/technetwork/middleware/oedq/downloads/index.html) for Linux x86-64-bit into folder `OEDQ/dockerfiles/12.2.1.3.0`.
+
+If a proxy is needed for the host to access yum.oracle.com during build, then first set up the appropriate environment, e.g.:
+
+        $ export http_proxy=myproxy.example.com:80
+        $ export https_proxy=myproxy.example.com:80
+        $ export no_proxy="localhost,127.0.0.1,localaddress,.localdomain.com"
```
Comment (Djelibeybi) [inline OracleEDQ/README.md]: I'm not sure why you're including `localaddress` and `.localdomain.com` in the `no_proxy` list.

File: OracleEDQ/README.md (hunk 4 above)
Comment (subhashsutrave) [reply OracleEDQ/README.md]: File EDQServer.env.list removed. As it is replaced by command line paramerters.

File: OracleEDQ/README.md (hunk 10 above)
Comment (subhashsutrave) [reply OracleEDQ/README.md]: Removed exporting no_proxy

File: OracleEDQ/README.md (hunk 9 above)
Comment (subhashsutrave) [reply OracleEDQ/README.md]: Fixed, changed to 
### Building the Oracle Java 8(Server JRE) Image
o build a EDQ image you start by building Oracle Java 8(Server JRE) and Oracle FMW Infrastructure images.

### Building the Oracle Java 8(Server JRE) Image

Download the Oracle Server JRE binary into folder `OracleJava/java-8` and build the image:

File: OracleEDQ/README.md (hunk 8 above)
Comment (subhashsutrave) [reply OracleEDQ/README.md]: Fixed. changed to 
To build a EDQ image you start by building Oracle Java 8(Server JRE) and Oracle FMW Infrastructure images.

File: OracleEDQ/12.2.1.3.0/container-scripts/CreateEDQDomain.py (hunk 7 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/container-scripts/CreateEDQDomain.py]: Used OracleFMWInfrastructure scripts as reference to create EDQ docker scripts. So used while block.

File: OracleEDQ/12.2.1.3.0/EDQServer.env.list (hunk 6 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/EDQServer.env.list]: File EDQServer.env.list removed. As it is replaced by command line paramerters.

File: OracleEDQ/12.2.1.3.0/EDQServer.env.list (hunk 5 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/EDQServer.env.list]: File EDQServer.env.list removed. As it is replaced by command line paramerters.

File: OracleEDQ/12.2.1.3.0/Dockerfile (hunk 3 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/Dockerfile]: Removed, commented directives

File: OracleEDQ/12.2.1.3.0/Dockerfile (hunk 2 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/Dockerfile]: Fixed, changed to 
#MAINTAINER Subhash Sutrave <subhash.sutrave@oracle.com>

File: OracleEDQ/12.2.1.3.0/Dockerfile (hunk 1 above)
Comment (subhashsutrave) [reply OracleEDQ/12.2.1.3.0/Dockerfile]: Fixed, Changed to
# Base image of this dockerfile is Oracle FMW Infrastructure and Oracle Java 8 docker image.

**Pull request merged**